torch/_prims/__init__.py

```diff
@@ -2963,4 +2967,87 @@ def _fft_c2r_aten(
     doc=_fft_c2r_doc,
 )
 
+
+#
+# Statically shaped dynamic operations
+#
+
+
+def _nonzero_static_meta(
+    a: TensorLikeType,
+    *,
+    size: int,
+    fill_value: int = -1,
+) -> TensorLikeType:
+    # Check if size is an non-negative integer
+    utils.check(
+        size >= 0,
+        lambda: f"'size' must be an non-negative integer but got {size}",
+        ValueError,
+    )
+
+    if size == 0:
+        # Return empty tensor
+        return torch.empty(0, dtype=torch.long)
+
+    shape: ShapeType = [size, a.dim()]
+    strides = utils.make_contiguous_strides_for(shape)
```

Comment:
regular `nonzero` returns discontiguous tensor (transposed, with strides (1,n)), it would be good for static nonzero to follow the pattern

Reply:
Any utils function to use, otherwise I guess I will just make the stride (1,n) directly

Reply:
there are a few ways to do it but you can either do `empty_permuted` or you can make a regular tensor and transpose it (slightly worse; empty_permuted is better.)